src/core/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils/hooks/useTheme';
+import { CommonProps } from '../utils/props';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import { IconButton } from '../Buttons';
+import { Tooltip } from '../Tooltip';
+import '@itwin/itwinui-css/css/side-navigation.css';
+
+export type SidebarProps = {
+  /**
+   * Buttons shown in the top portion of sidebar.
+   * Recommended to use `SidebarButton` components.
+   */
+  mainItems: React.ReactNode[];
+  /**
+   * Buttons shown at the bottom of sidebar.
+   */
+  secondaryItems?: React.ReactNode[];
+  /**
+   * Control the visibility of "expander" icon button.
+   * @default 'top'
+   */
+  expanderVisibility?: 'top' | 'bottom' | 'hidden';
+  /**
+   * Controlled flag to expand/collapse the sidebar.
+   */
+  isExpanded?: boolean;
+  /**
+   * Callback fired when the "expander" icon is clicked.
+   */
+  onExpanderClick?: () => void;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Sidebar component.
+ * @example
+ * <Sidebar
+ *  mainItems={[
+ *    <SidebarButton startIcon={<SvgPlaceholder />}>App 1</SidebarButton>,
+ *    <SidebarButton startIcon={<SvgPlaceholder />}>App 2</SidebarButton>,
+ *    <SidebarButton startIcon={<SvgPlaceholder />}>App 3</SidebarButton>,
+ *  ]}
+ * secondaryItems={[
+ *  <SidebarButton
+ * ]}
+ * />
+ */
+export const Sidebar = (props: SidebarProps) => {
+  const {
+    mainItems,
+    secondaryItems,
+    expanderVisibility = 'top',
+    className,
+    isExpanded,
```

Comment:
or just give default value and no need to check for undefined

Reply:
Initially I wanted to differentiate between controlled prop and internal state so I left it undefined